habitat_baselines/common/base_il_trainer.py

```diff
@@ -0,0 +1,507 @@
+        self.config = config
+        self._flush_secs = 30
+        self._make_dirs()
+
+    @property
+    def flush_secs(self):
+        return self._flush_secs
+
+    @flush_secs.setter
+    def flush_secs(self, value: int):
+        self._flush_secs = value
+
+    def _make_dirs(self):
+        self._make_log_dir()
+        self._make_ckpt_dir()
+        if self.config.EVAL_SAVE_RESULTS:
+            self._make_results_dir()
+
+    def _make_log_dir(self):
+        if self.config.LOG_METRICS:
+            if not os.path.isdir(self.config.OUTPUT_LOG_DIR):
+                os.makedirs(self.config.OUTPUT_LOG_DIR)
+
+    def _make_ckpt_dir(self):
+        if not os.path.isdir(self.config.CHECKPOINT_FOLDER):
+            os.makedirs(self.config.CHECKPOINT_FOLDER)
+
+    def _make_results_dir(self):
+        if self.config.TRAINER_NAME == "eqa-cnn-pretrain":
+            for type in ["rgb", "seg", "depth"]:
+                dir_name = self.config.RESULTS_DIR.format(
+                    split="val", type=type
+                )
+                if not os.path.isdir(dir_name):
+                    os.makedirs(dir_name)
+        else:
+            dir_name = self.config.RESULTS_DIR.format(split="val")
+            if not os.path.isdir(dir_name):
+                os.makedirs(dir_name)
+
+    def train(self) -> None:
+        raise NotImplementedError
+
+    def eval(self) -> None:
+        r"""Main method of trainer evaluation. Calls _eval_checkpoint() that
+        is specified in Trainer class that inherits from BaseILTrainer
+
+        Returns:
+            None
+        """
+        self.device = (
+            torch.device("cuda", self.config.TORCH_GPU_ID)
+            if torch.cuda.is_available()
+            else torch.device("cpu")
+        )
+
+        with TensorboardWriter(
+            self.config.TENSORBOARD_DIR, flush_secs=self.flush_secs
+        ) as writer:
+            if os.path.isfile(self.config.EVAL_CKPT_PATH_DIR):
+                # evaluate singe checkpoint
+                self._eval_checkpoint(self.config.EVAL_CKPT_PATH_DIR, writer)
+            else:
+                # evaluate multiple checkpoints in order
+                prev_ckpt_ind = -1
+                while True:
+                    current_ckpt = None
+                    while current_ckpt is None:
+                        current_ckpt = poll_checkpoint_folder(
+                            self.config.EVAL_CKPT_PATH_DIR, prev_ckpt_ind
+                        )
+                        time.sleep(2)  # sleep for 2 secs before polling again
+                    logger.info(f"=======current_ckpt: {current_ckpt}=======")
+                    prev_ckpt_ind += 1
+                    self._eval_checkpoint(
+                        checkpoint_path=current_ckpt,
+                        writer=writer,
+                        checkpoint_index=prev_ckpt_ind,
+                    )
+
+    def _setup_eval_config(self, checkpoint_config: Config) -> Config:
```

Comment:
Same comment as with `eval`

Reply:
I have moved `eval` and `setup_eval_config` to the `BaseTrainer` in the latest commit.